### DISCUSSION TITLE
Please tell me some good information to help me with coding.

### DISCUSSION THREAD

**Discussion by happy-spark** [community/community#40000] [Other Feature Feedback, Questions, & Ideas] [answered]
Please tell me some good information to help me with coding.

**Comment by CBID2** [comment, marked as answer]
Hi @happy-spar! :) Can you elaborate on your question please?

**Reply by happy-spark** [reply]
I'm a beginner programmer. Therefore, I would like to get recommendations on how to learn coding

**Reply by CBID2** [reply]
I see! I highly recommend learning HTML, CSS, and JavaScript before moving on to other programming languages. They are the key foundations of web development. [Scrimba](https://scrimba.com/about) has great free courses about these languages. If you want to go pro, check out their [scholarship](https://scholarship.scrimba.com/) for the boot camp. I hope this helps you @happy-spark!

**Reply by happy-spark** [reply]
Thank you very much I was studying HTML CSS Javascript these days and it was helpful

**Reply by CBID2** [reply]
Oh that's wonderful @happy-spark! :)

**Comment by jellybone** [comment]
ask chat gpt they give more details

**Reply by CBID2** [reply]
Ehh not all the time. Sometimes, ChatGPT spits out inaccurate code.

**Reply by Aadi-stack** [reply]
yes @CBID2 right, sometime chatbot does not accurate result

**Comment by zenodegenkamp** [comment]
I would highly recommand to join a free/paid course to learn how to code, this will help you with a solid basis. Free options are wildly available on youtube. A paid option I myself used is Scrimba: Scrimba stands out because it's not just about watching videos; it's about interaction. You can pause tutorials and mess around with the code right there in the browser. It's like learning to cook by actually cooking, not just reading recipes. This hands-on approach is golden for cementing concepts. Good luck on your coding journey!

**Reply by happy-spark** [reply]
Thanks for letting me know!

**Comment by abhiramasonny** [comment]
I started with Scratch, moved on to HTML and JS, and then transitioned over to Python. when you're on Python, I would recommend using as few libraries as possible until you feel comfortable with how to solve problems, and have a problem-solving mind. when you feel that you are ready, learn Numpy, and OpenCV2. You can then think about what you want to do as a software dev and move on from there.

**Reply by happy-spark** [reply]
Wow, your programming roadmap is very similar to mine. Thanks for letting me know!

**Reply by abhiramasonny** [reply]
No problem! good luck on learning
